src/critics/obstacles_critic.cpp

```diff
@@ -1,99 +1,69 @@
 {
-  constexpr double collision_cost_value = std::numeric_limits<double>::max() / 2;
-
-
-  enum TrajectoryState : uint8_t { Collision, Inflated, Free };
+  constexpr double COLLISION_COST = std::numeric_limits<double>::max() / 4;
 
   for (size_t i = 0; i < trajectories.shape()[0]; ++i) {
-    double min_dist_to_obstacle = std::numeric_limits<double>::max();
-    TrajectoryState state = Free;
+    bool trajectory_collide = false;
 
+    unsigned char trajectory_cost = nav2_costmap_2d::FREE_SPACE;
+    auto trajectory = xt::view(trajectories, i, xt::all(), xt::all());
     for (size_t j = 0; j < trajectories.shape()[1]; ++j) {
       auto point = xt::view(trajectories, i, j, xt::all());
-      unsigned char cost = costAtPose(point);
-      if (!isFree(cost)) {
-        if (inCollision(cost)) {
-          state = Collision;
-          break;
-        }
-        state = Inflated;
-        min_dist_to_obstacle = std::min(toDist(cost), min_dist_to_obstacle);
+      unsigned char pose_cost = costAtPose(point);
+      trajectory_cost = std::max(trajectory_cost, pose_cost);
+
+      if (inCollision(pose_cost)) {
+        trajectory_collide = true;
+        break;
       }
     }
 
-    if (state == Inflated) {
-      costs[i] += scoreDistance(min_dist_to_obstacle);
-    } else if (state == Collision) {
-      costs[i] += collision_cost_value;
-    }
+    costs[i] += trajectory_collide ? COLLISION_COST : scoreCost(trajectory_cost);
   }
 }
 
-unsigned char ObstaclesCritic::costAtPose(const auto & point)
-{
-  unsigned char cost;
-
-  if (consider_footprint_) {
-    cost = static_cast<unsigned char>(collision_checker_.footprintCostAtPose(
-      point(0), point(1), point(2), costmap_ros_->getRobotFootprint()));
-  } else {
-    cost = static_cast<unsigned char>(collision_checker_.pointCost(point(0), point(1)));
-  }
-
-  return cost;
-}
-
-bool ObstaclesCritic::inCollision(unsigned char & cost) const
+bool ObstaclesCritic::inCollision(unsigned char cost) const
 {
-  unsigned char max_valid_cost;
-  if (consider_footprint_) {
-    max_valid_cost = nav2_costmap_2d::LETHAL_OBSTACLE;
-  } else {
-    max_valid_cost = nav2_costmap_2d::INSCRIBED_INFLATED_OBSTACLE;
+  bool is_tracking_unknown = costmap_ros_->getLayeredCostmap()->isTrackingUnknown();
+
+  switch (cost) {
+  case (nav2_costmap_2d::LETHAL_OBSTACLE):
+    return true;
+  case (nav2_costmap_2d::INSCRIBED_INFLATED_OBSTACLE):
+    return consider_footprint_ ? false : true;
+  case (nav2_costmap_2d::NO_INFORMATION):
+    return is_tracking_unknown ? false : true;
+  default:
+    return false;
   }
-
-  if (costmap_ros_->getLayeredCostmap()->isTrackingUnknown()) {
-    return cost >= max_valid_cost && cost != nav2_costmap_2d::NO_INFORMATION;
-  }
-
-  return cost >= max_valid_cost;
-}
-
-bool ObstaclesCritic::isFree(unsigned char & cost) const
-{
-	return cost == nav2_costmap_2d::FREE_SPACE;
 }
 
-double ObstaclesCritic::toDist(unsigned char & cost)
+double ObstaclesCritic::toDist(unsigned char cost)
```

Comment:
This function can be deleted since unused

Reply:
removed